src/renderer/src/components/Button/index.tsx

```diff
@@ -0,0 +1,70 @@
+/* eslint-disable no-nested-ternary */
+import { ButtonProps, CircularProgress, SvgIconTypeMap } from '@mui/material'
+import { OverridableComponent } from '@mui/material/OverridableComponent'
+import { StyledButton } from './styles'
+
+type IconType = OverridableComponent<SvgIconTypeMap> | null
+
+type ButtonPropTypesHelper = ButtonProps & {
+  children: React.ReactNode
+  icon?: IconType
+  loading?: boolean
+  disabled?: boolean
```

Comment:
You can take out the these three props and their associated use as they won't be used here